extensions/nns/src/main.rs

```diff
@@ -15,6 +17,11 @@ pub struct NnsOpts {
     /// `dfx nns` subcommand arguments.
     #[clap(subcommand)]
     subcmd: SubCommand,
+
+    // global args have to be wrapped with Option for now: https://github.com/clap-rs/clap/issues/1546
+    /// Path to cache of DFX which executed this extension.
+    #[arg(long, global = true)]
+    dfx_cache_path: Option<String>,
```

Comment:
This will simplify things below a little:
```suggestion
    dfx_cache_path: Option<PathBuf>,
```